### PR TITLE
[Android Editor] Add support for launching the Play window in PiP mode

### LINKED ISSUE #77966: Android editor freezes after script error in project due to not switching to the editor view automatically
### Godot version

Mobile Editor  3.6.beta.2 android

### System information

I am on a samsung a04s android 11

### Issue description

When you run a piece of code that has a error in it in the mobile version of godot the scene becomes black and it also does not let you leave the scene the only way to go back is by manualy leaveing the app i havd no idea what causes this or how to fix it but it would be nice if you would i have also provided a little demostration down below kind regards

:  random poor  kid

### Steps to reproduce

Go into godot type a error and run it should brick the entire thing 

### Minimal reproduction project

Screenshot_20230607-172012_Godot Editor 3
Screenshot_20230607-172021_Godot Editor 3

### PR DESCRIPTION
**Pull request by m4gr3d**
Adds support for launching the Play window in picture-in-picture (PiP) mode, which provides additional flexibility when developing and running the game at the same time especially for phones with limited screen estate.

https://github.com/user-attachments/assets/4258934a-1946-4557-a28a-982bba4d9e75

**Note:**
- The Play window in PiP mode does not take touch inputs; interaction using touch inputs requires restoring the Play window to full screen.
- When PiP is enabled, running the game with breakpoints enabled will launch directly in PiP mode

https://github.com/user-attachments/assets/f1e1d221-8a3a-4e78-a218-4a52893d0230

- Fixes https://github.com/godotengine/godot/issues/77966
  - When the Play window freezes because of a script error in the project, the PiP button can be used to minimize the Play window and access the editor

https://github.com/user-attachments/assets/f4b45f12-7a3b-4544-926f-e4356b069488

### CODE REVIEW

File: doc/classes/EditorSettings.xml [hunk 1]
```diff
@@ -961,7 +961,17 @@
...
+			- [b]Auto (based on screen size)[/b] (default) will automatically choose how to launch the Play window based on the device and screen metrics. Defaults to [b]Same as Editor[/b] on phones and [b]Side-by-side with Editor[/b] on tablets.
+			- [b]Same as Editor[/b] will launch the Play window in the same window as the Editor.
+			- [b]Side-by-side with Editor[/b] will launch the Play window side-by-side with the Editor window.
+			[b]Note:[/b] Only available in the Android editor.
+		</member>
+		<member name="run/window_placement/play_window_pip_mode" type="int" setter="" getter="">
+			Specifies the picture-in-picture (PiP) mode for the Play window.
+			- [b]Disabled[/b] (default for non-Android editor) PiP is disabled for the Play window.
+			- [b]Enabled[/b] If the device supports it, PiP is always enabled for the Play window. The Play window will contain a button to enter PiP mode.
+			- [b]Enabled when Play window is same as Editor[/b] (default for Android editor) If the device supports it, PiP is enabled when the Play window is the same as the Editor. The Play window will contain a button to enter PiP mode.
```
Comment (akien-mga) [inline doc/classes/EditorSettings.xml:974]:
```suggestion
			- [b]Disabled[/b] (default for non-Android editor): PiP is disabled for the Play window.
			- [b]Enabled:[/b] If the device supports it, PiP is always enabled for the Play window. The Play window will contain a button to enter PiP mode.
			- [b]Enabled when Play window is same as Editor[/b] (default for Android editor): If the device supports it, PiP is enabled when the Play window is the same as the Editor. The Play window will contain a button to enter PiP mode.
```
That being said:

> [b]Disabled[/b] (default for non-Android editor): PiP is disabled for the Play window.

seems to contradict:

> [b]Note:[/b] Only available in the Android editor.

So maybe that parenthesis should be removed.

**Review by akien-mga — APPROVED** [review]

File: doc/classes/EditorSettings.xml (hunk 1 above)
Comment (m4gr3d) [reply doc/classes/EditorSettings.xml]: Done!

**Comment by akien-mga** [comment]
Thanks!

**Comment by arutemu** [comment]
I am sorry if it's not a good place to ask questions, especially since the changes are already merged.
But is there an option to launch the game in PiP mode by default? (Which would be one of the most useful launch options)

Despite a number of available setting options,  none seems to be doing this.

**Comment by m4gr3d** [comment]
> I am sorry if it's not a good place to ask questions, especially since the changes are already merged. But is there an option to launch the game in PiP mode by default? (Which would be one of the most useful launch options)
> 
> Despite a number of available setting options, none seems to be doing this.

@arutemu I had considered this but at the time couldn't settle on the proper UX to make this happen. Thinking on it some more, launching in PiP mode could be an option in the `run/window_placement/android_window` editor setting.

I'll push a PR with those changes.

**Comment by arutemu** [comment]
@m4gr3d thank you for the update. Would be glad to test it :)

Just to clarify - such feature may not be useful on the devices with small displays,  but could be really great on android tablets.
Especially since android's implementation of the windowed apps is quite... not polished, to be honest. 

Also,  small question - is it fine to reach with similar QoL/usability ideas via github comments,  or is there some more suitable place?

**Comment by m4gr3d** [comment]
> Just to clarify - such feature may not be useful on the devices with small displays, but could be really great on android tablets.
Especially since android's implementation of the windowed apps is quite... not polished, to be honest.

Note that for tablets, you can also enable the `Side-by-side with Editor` option in the editor settings under the `Run` -> `Window Placement` -> `Android Window` section.

> Also, small question - is it fine to reach with similar QoL/usability ideas via github comments, or is there some more suitable place?

If the comments are within the scope to the PR being merged, then posting on the PR is fine.

For bug reports, you should create a new issue through the [project issue page](https://github.com/godotengine/godot/issues).
For new feature proposals, you should submit feature proposals on the [Godot proposals repository](https://github.com/godotengine/godot-proposals).

Hope that helps!

**Comment by Davinci4** [comment]
Is there any chance that you Could you sent me the apk with the commuts you made ?

**Comment by m4gr3d** [comment]
> Is there any chance that you Could you sent me the apk with the commuts you made ?

@Davinci4 these changes are available in the latest 4.4 dev2 builds. You can get access to that build from the play store through https://play.google.com/apps/testing/org.godotengine.editor.v4.

In addition the apks for each release can be found in https://github.com/godotengine/godot-builds/releases.

And lastly the editor apks for each PR can be found under the `checks` tab for that PR.

**Pull request merged**